File: php/fizzbuzz/tests/FizzBuzzTest.php
<?php

require_once 'PHPUnit/Framework.php';

require_once dirname(__FILE__) . '/../FizzBuzz.php';

class FizzBuzzTest extends PHPUnit_Framework_TestCase {

    protected function setUp() {
        $this->object = new Fizzbuzz;
    }

    public function testFizzBuzzValueOfOneShouldBeOne() {
        $this->assertEquals("1", $this->object->getFizzBuzzValueOf(1));
    }

    public function testFizzBuzzValueOfTwoShouldBeTwo() {
        $this->assertEquals("2", $this->object->getFizzBuzzValueOf(2));
    }

    public function testFizzBuzzValueOfThreeShouldBeFizz() {
        $this->assertEquals("Fizz", $this->object->getFizzBuzzValueOf(3));
    }

    public function testFizzBuzzValueofFiveShouldBeBuzz() {
        $this->assertEquals("Buzz", $this->object->getFizzBuzzValueOf(5));
    }

    public function testFizzBuzzMultipleValueOfThreeShouldBeFizz() {
        $this->assertEquals("Fizz", $this->object->getFizzBuzzValueOf(6));
    }

    public function testFizzBuzzMultipleValueOfFiveShouldBeBuzz() {
        $this->assertEquals("Buzz", $this->object->getFizzBuzzValueOf(10));
    }

    public function testFizzBuzzMultipleValueOfFiveAndThreeShouldBeFizzBuzz() {
        $this->assertEquals("FizzBuzz", $this->object->getFizzBuzzValueOf(15));
    }

}
?>

File: java/fizzbuzz/test/fizzbuzz/FizzBuzzTest.java
package fizzbuzz;

import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

public class FizzBuzzTest {

    private FizzBuzz _fizzbuzz;

    @Before
    public void setUp() {
       _fizzbuzz = new FizzBuzz();
    }

    @Test
    public void testFizzBuzzValueOfOneShouldBeOne() {
        assertEquals("1", this._fizzbuzz.getFizzBuzzValueOf(1));
    }
    
    @Test
    public void testFizzBuzzValueOfTwoShouldBeTwo() {
        assertEquals("2", this._fizzbuzz.getFizzBuzzValueOf(2));
    }

    @Test
    public void testFizzBuzzValueOfThreeShouldBeFizz() {
        assertEquals("Fizz", this._fizzbuzz.getFizzBuzzValueOf(3));
    }

    @Test
    public void testFizzBuzzValueofFiveShouldBeBuzz() {
        assertEquals("Buzz", this._fizzbuzz.getFizzBuzzValueOf(5));
    }

    @Test
    public void testFizzBuzzMultipleValueOfThreeShouldBeFizz() {
        assertEquals("Fizz", this._fizzbuzz.getFizzBuzzValueOf(6));
    }

    @Test
    public void testFizzBuzzMultipleValueOfFiveShouldBeBuzz() {
        assertEquals("Buzz", this._fizzbuzz.getFizzBuzzValueOf(10));
    }

    @Test
    public void testFizzBuzzMultipleValueOfFiveAndThreeShouldBeFizzBuzz() {
        assertEquals("FizzBuzz", this._fizzbuzz.getFizzBuzzValueOf(15));
    }

}